core/src/main/java/org/elasticsearch/index/query/AndQueryParser.java

```diff
@@ -72,21 +72,11 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 } else if (parseContext.isDeprecatedSetting(currentFieldName)) {
                     // skip
                 } else if (token == XContentParser.Token.START_ARRAY) {
-                    if ("filters".equals(currentFieldName)) {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
-                        }
-                    } else {
-                        queriesFound = true;
-                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-                            Query filter = parseContext.parseInnerFilter();
-                            if (filter != null) {
-                                queries.add(filter);
-                            }
+                    queriesFound = true;
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                        Query filter = parseContext.parseInnerFilter();
+                        if (filter != null) {
+                            queries.add(filter);
```

Comment:
@javanna Just came back to your comment here, what do you mean by "more lenient"? I think this change basically allows anything inside an array to be parsed as inner queries, as far as I see this was the behaviour of the `else` branch before. Maybe I missunderstood your comment.